opentelemetry/src/sdk/trace/tracer.rs

```diff
@@ -186,9 +184,8 @@ impl crate::trace::Tracer for Tracer {
         };
 
         // Build context for sampling decision
-        let (no_parent, trace_id, parent_span_id, remote_parent, parent_trace_flags) = parent_span
-            .as_ref()
-            .map(|parent| {
+        let (no_parent, trace_id, parent_span_id, remote_parent, parent_trace_flags) =
```

Comment:
Maybe use `match` directly here instead of the `if let Some(_) .. { } else { }`? You could add a `.map(|parent| parent.span_context())` to the match expression to make it more concise.

It would be nice to get rid of the 5-tuple, too. Since `no_parent` and `remote_parent` are just `bool`s, maybe initialize them to the non-parent case before the match? Can probably do the same for `parent_span_id`, since initializing to the const `SpanId::INVALID` should be cheap enough.

Reply:
sounds good, definitely nice to not have the 5-tuple 👍